packages/block-library/src/cover/edit.js

```diff
@@ -311,11 +314,43 @@ function CoverEdit( {
 		minHeight,
 		minHeightUnit,
 		style: styleAttribute,
-		url,
 		alt,
 		allowedBlocks,
 		templateLock,
 	} = attributes;
+
+	let { url, backgroundType } = attributes;
+
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const media = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+	const mediaUrl = media?.source_url;
+
+	useEffect( () => {
+		if ( useFeaturedImage ) {
+			setAttributes( {
+				dimRatio: dimRatio === 100 ? 50 : dimRatio,
+			} );
+		}
+	}, [ useFeaturedImage ] );
+
+	// hijack the url for edit preview
+	// of the dynamic featured image
+	if ( useFeaturedImage && mediaUrl ) {
+		url = mediaUrl;
+		backgroundType = IMAGE_BACKGROUND_TYPE;
+	}
```

Comment:
Nit: maybe we could avoid the destructuring of attributes and only define these variables here instead of hijacking them.

```
const url = useFeaturedImage ? mediaUrl : attributes.url;
const backgroundType = useFeaturedImage ? IMAGE_BACKGROUND_TYPE : attributes.backgroundType;
```